srtcore/epoll.cpp

```diff
@@ -636,6 +712,90 @@ int CEPoll::wait(const int eid, set<SRTSOCKET>* readfds, set<SRTSOCKET>* writefd
     return 0;
 }
 
+int CEPoll::swait(CEPollDesc& d, map<SRTSOCKET, int>& st, int64_t msTimeOut, bool report_by_exception)
```

Comment:
This is a new version of `CEPoll::uwait`, which works much like the latter, but it uses different signature and different rules for returned value. This is modeled largely after `uwait` and in the implementation there should be not much differences.